packages/react/src/file-input/FileInput.tsx

```diff
@@ -139,83 +138,76 @@ export const FileInput = forwardRef<HTMLInputElement, FileInputProps>(
 		const fallbackHint =
 			hint || (accept && `Files accepted: ${acceptedFileExtensions}`);
 
+		const required = Boolean(requiredProp);
+		const buttonLabel = `Select ${multiple ? 'files' : 'file'}`;
+		const ariaLabel = [
+			buttonLabel,
+			label,
+			required && 'required',
+			invalid && 'invalid',
+			getSelectedFilesMessage({ fileNames, multiple }),
+		]
+			.filter(Boolean)
+			.join(', ');
+
 		return (
 			<Field
 				hideOptionalLabel={hideOptionalLabel}
 				hint={fallbackHint}
-				id={inputId}
+				id={id}
 				invalid={invalid}
 				label={label}
 				message={message}
-				required={Boolean(required)}
+				required={required}
 			>
-				{(a11yProps) => {
-					const visibleButtonLabel = `Select file${multiple ? 's' : ''}`;
-					const ariaLabel = [
-						visibleButtonLabel,
-						label,
-						a11yProps['aria-required'] && 'required',
-						a11yProps['aria-invalid'] && 'invalid',
-						getSelectedFilesMessage({ fileNames, multiple }),
-					]
-						.filter(Boolean)
-						.join(', ');
-
-					return (
-						<>
-							<Stack gap={1} justifyContent="flex-start" alignItems="start">
-								<Button
-									{...a11yProps}
-									aria-invalid={undefined} // Buttons not announced as invalid when `aria-invalid`, add announcement to label instead
-									aria-label={ariaLabel}
-									aria-required={undefined} // Buttons can't be `aria-required`, add announcement to label instead
```

Comment:
I think there's value in these comments because we're doing something unorthodox.

Reply:
Agreed. Perhaps I'll add a comment saying that we're specifically only applying certain props since it feels weirder to set and then unset

Reply:
Yeah, for sure. Even something about the hidden semantic input?

Reply:
Updated this and all other comments in 5673434454d14fe1c0f074e6c8dceb90f240ca7d